### PR TITLE
[RFC] Expression functions as generic formal function parameters

### PR DESCRIPTION
**Pull request by clairedross**

### CODE REVIEW

**Comment by djthyrax** [comment]
I really like this proposal, having had the need of something similar a couple of days ago.

**Comment by raph-amiard** [comment]
Thanks guys for your feedback ! @djthyrax @AdaDoom3 could you please upvote the OP ? it will make it easier for us to track support. TIA!

File: considered/rfc-expression-functions-as-default-for-generic-formal-function-parameters.rst [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+generic
+   type T is private;
+package Stacks is
+   type Stack is private;
+   procedure Push (S : Stack; X : T);
+   ...
+
+However, if we use this generic to store access based structure, it may not do
+what we expect:
```
Comment (vser1) [inline considered/rfc-expression-functions-as-default-for-generic-formal-function-parameters.rst:52]: Is the repetition of `what we expect` on purpose?

Comment (yannickmoy) [reply considered/rfc-expression-functions-as-default-for-generic-formal-function-parameters.rst:52]: That's not what we expect indeed. ;-)
I've submitted a typo-fixing PR, thanks.

**Pull request merged**